### PR TITLE
Support for offsetdelete

### LINKED ISSUE #1009: Add support for offsetdelete
Add support for offsetdelete: https://kafka.apache.org/protocol.html#The_Messages_OffsetDelete

### PR DESCRIPTION
**Pull request by a-dot**
Closes #1009

The tests need kafka 2.4 to successfully run.

### CODE REVIEW

**Comment by rhansen2** [comment]
Hi @a-dot,

Thanks for the contribution! You can use a construct like https://github.com/segmentio/kafka-go/blob/d8dd9b00cf48af7abc5144bbc574ccf03c3199a0/createpartitions_test.go#L11 to gate your tests around the proper kafka versions.

**Comment by a-dot** [comment]
> Hi @a-dot,
> 
> Thanks for the contribution! You can use a construct like
> 
> https://github.com/segmentio/kafka-go/blob/d8dd9b00cf48af7abc5144bbc574ccf03c3199a0/createpartitions_test.go#L11
> 
> to gate your tests around the proper kafka versions.

Awesome! Thanks for the tip. I fixed that.

**Comment by a-dot** [comment]
Any hints as to why the tests fail on kafka-271 would be appreciated. Unless I'm mistaken, it doesn't look like it's my test that is failing?

File: offsetdelete.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/segmentio/kafka-go/protocol/offsetdelete"
+)
+
+// OffsetDelete deletes the offset for a consumer group on a particular topic
+// at a particular partition.
```
Comment (rhansen2) [inline offsetdelete.go:13]:
```suggestion
// for a particular partition.
```

File: protocol/offsetdelete/offsetdelete_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package offsetdelete_test
+
+import (
+	"testing"
+
+	"github.com/segmentio/kafka-go/protocol/offsetdelete"
+	"github.com/segmentio/kafka-go/protocol/prototest"
+)
+
+func TestOffsetDeleteequest(t *testing.T) {
```
Comment (rhansen2) [inline protocol/offsetdelete/offsetdelete_test.go:10]: Any reason not to add a similar test for the response type?

**Comment by rhansen2** [comment]
The test failure looked like some unrelated flakiness. I re-ran the failed job and it looks good.

File: protocol/offsetdelete/offsetdelete_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (a-dot) [reply protocol/offsetdelete/offsetdelete_test.go]: Fixed in [0db9857](https://github.com/segmentio/kafka-go/pull/1010/commits/0db9857b3a9812cd4cff914400aff47b0501d164)

Comment (a-dot) [reply protocol/offsetdelete/offsetdelete_test.go]: Fixed in [0db9857](https://github.com/segmentio/kafka-go/pull/1010/commits/0db9857b3a9812cd4cff914400aff47b0501d164)

File: offsetdelete.go (hunk 1 above)
Comment (a-dot) [reply offsetdelete.go]: Fixed [5ff0c69](https://github.com/segmentio/kafka-go/pull/1010/commits/5ff0c691f41bfc6e133af496c3d28875d74a4d65)

**Review by rhansen2 — APPROVED** [review]
LGTM!

**Pull request merged**